src/modals/SendModal/index.tsx

```diff
@@ -59,40 +74,59 @@ const SendModal = ({ onClose }: SendModalProps) => {
     currentNetwork
   } = useGlobalContext()
   const { setAddress } = useAddressesContext()
+  const { dappTransactionData, requestEvent, walletConnect } = useWalletConnectContext()
   const [title, setTitle] = useState('')
-  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>()
+  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>(dappTransactionData)
   const [isLoading, setIsLoading] = useState(false)
-  const [step, setStep] = useState<StepIndex>(1)
+  const [step, setStep] = useState<StepIndex>(StepIndex.Send)
   const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
   const [consolidationRequired, setConsolidationRequired] = useState(false)
   const [isSweeping, setIsSweeping] = useState(false)
   const [sweepUnsignedTxs, setSweepUnsignedTxs] = useState<SweepAddressTransaction[]>([])
   const [unsignedTxId, setUnsignedTxId] = useState('')
   const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [contractAddress, setContractAddress] = useState('')
   const [fees, setFees] = useState<bigint>()
   const theme = useTheme()
 
   useEffect(() => {
-    if (step === 1) {
-      setTitle('Send')
-    } else if (step === 2) {
-      setTitle('Info Check')
-    } else if (step === 3) {
-      setTitle('Password Check')
-    }
+    setTitle(stepToTitle.get(step) || 'Unknown')
   }, [setStep, setTitle, step])
 
   const confirmPassword = () => {
     if (consolidationRequired) setIsConsolidateUTXOsModalVisible(false)
-    setStep(3)
+    setStep(StepIndex.PasswordCheck)
   }
 
   const buildTransaction = async (transactionData: SendTransactionData) => {
-    setTransactionData(transactionData)
+    const { fromAddress, toAddress, gasAmount, gasPrice, contractCode, contractState, issueTokenAmount, script } =
+      transactionData
+    let { amount } = transactionData
+
+    const isContractTx = contractCode !== ''
+    const isScriptTx = script !== ''
```

Comment:
If we decide to be a bit more strict with Tx Types (as suggested above, defining `ContractTx`, `TransferTx` etc.), we could take the opportunity to define some [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), like those `isContractTx` and `isScriptTx`.

Reply:
This is a bit less trivial than it appears: pending tx is for any type of pending tx, where some properties are optional. There's going to be a bunch of refactoring that'll be needed. Let's push it to another PR.